modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/Compiler.java

```diff
@@ -78,23 +79,31 @@ public Compiler(CompilerContext context) {
 
     public void compile(String sourcePkg) {
         loadBuiltInPackage();
-        switch (compilerPhase) {
-            case DEFINE:
-                define(sourcePkg);
-                break;
-            case TYPE_CHECK:
-                typeCheck(define(sourcePkg));
-                break;
-            case CODE_ANALYZE:
-                codeAnalyze(typeCheck(define(sourcePkg)));
-                break;
-            case DESUGAR:
-                desugar(codeAnalyze(typeCheck(define(sourcePkg))));
-                break;
-            default:
-                gen(desugar(codeAnalyze(typeCheck(define(sourcePkg)))));
-                break;
+        if (this.stopCompilation(CompilerPhase.DEFINE)) {
```

Comment:
So now aren't we considering options compilerphase?

Reply:
Added it to the stopCompilation() method.